File: apps/agor-android/app/src/main/kotlin/live/agor/app/models/Worktree.kt
package live.agor.app.models

import androidx.compose.runtime.Immutable
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Immutable
@Serializable
data class Worktree(
    @SerialName("worktree_id") val worktreeId: String,
    @SerialName("repo_id") val repoId: String,
    @SerialName("board_id") val boardId: String? = null,
    val name: String,
    val branch: String? = null,
    val path: String? = null,
    val status: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
    @SerialName("created_by") val createdBy: String? = null,
    val archived: Boolean? = null,
    @SerialName("archived_reason") val archivedReason: String? = null,
    @SerialName("others_can") val othersCan: String? = null,
)

File: apps/agor-rust-android/src/models/worktree.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct Worktree {
    pub worktree_id: String,
    pub repo_id: String,
    pub board_id: Option<String>,
    pub name: String,
    pub branch: Option<String>,
    pub path: Option<String>,
    pub status: Option<String>,
    pub created_at: Option<String>,
    pub created_by: Option<String>,
    pub archived: Option<bool>,
    pub archived_reason: Option<String>,
    pub others_can: Option<String>,
}
